libraries/websphere_base.rb

```diff
@@ -630,6 +630,25 @@ def deploy_to_cluster(appl_name, appl_file, clus_name, map_web_mod_to_vh = nil)
         wsadmin_exec("wsadmin deploy #{appl_file} to cluster #{clus_name}", cmd)
         save_config
       end
+
+      # Retrieve those cluster templates already defined for a cluster
+      def get_cluster_templates(clus_name)
+        cmd = "-c \"AdminTask.listClusterMemberTemplates('[-clusterName #{clus_name}]')\""
+        mycmd = wsadmin_returns(cmd)
+        # example output:'V7MemberTemplate(templates/clusters/sample-cluster/servers/V7MemberTemplate|server.xml#Server_1506097558200)\nV7MemberTemplate0(templates/clusters/sample-cluster/servers/V7MemberTemplate0|server.xml#Server_1508144541044)'
+        was_templates = []
+        return was_templates if mycmd.error?
+        str = mycmd.stdout.chomp.match(/'(.*?)'/) # get only what's between ''.
+        return was_templates if str.nil?
+        str_match = str.captures.first
+        # get the first match only, removing outer brackets ['']
+        raw_servers = str_match.split("\n")
+        raw_servers.each do |s|
+          was_template = s.split('(')[0]
+          was_templates << was_template
+        end
+        Chef::Log.debug("get_cluster_templates() result #{was_templates}")
```

Comment:
As you don't explicitly return was_templates, I think the `Chef::Log.debug` will make the function return `nil`.

```
[1] pry(main)> def testfunction()
[1] pry(main)*   a='a'
[1] pry(main)*   print 'test'
[1] pry(main)* end
=> :testfunction
[2] pry(main)> testfunction()
test=> nil
[3] pry(main)> def testfunction()
[3] pry(main)*   a='a'
[3] pry(main)* end
=> :testfunction
[4] pry(main)> testfunction()
=> "a"
```